fabrics.c

```diff
@@ -657,10 +658,8 @@ int nvmf_discover(const char *desc, int argc, char **argv, bool connect)
 out_free_ctrl:
 	nvme_free_ctrl(c);
 out_free:
-	if (hnqn)
-		free(hnqn);
-	if (hid)
-		free(hid);
+	free(hostnqn);
+	free(hostid);
```

Comment:
This is wrong. 'hostnqn' and/or 'hostid' might point to a command-line argument, in which case it can't be freed.

Reply:
Oh, I see it's `argconfig_parse()`. I really start to develop a very strong opinion on this parser.